pandas/tests/groupby/test_categorical.py

```diff
@@ -737,8 +737,9 @@ def test_preserve_on_ordered_ops(func, values):
     df = pd.DataFrame({"payload": [-1, -2, -1, -2], "col": c})
     g = df.groupby("payload")
     result = getattr(g, func)()
+    expected_col = pd.Categorical(values, ordered=True)
     expected = pd.DataFrame(
-        {"payload": [-2, -1], "col": pd.Series(values, dtype=c.dtype)}
+        {"payload": [-2, -1], "col": expected_col}
```

Comment:
I think the output of this one was wrong, the aggregated result should not preserve its original categories?

currently on master, this will return `CategoricalDtype(categories=['first', 'fourth', 'second', 'third'], ordered=True)`, but i think this should be `CategoricalDtype(categories=['second', 'third'], ordered=True)`

@jbrockmendel